packages/javascript/jupyterlab-plotly/src/Figure.ts

```diff
@@ -939,6 +942,21 @@ export class FigureView extends DOMWidgetView {
     }
   }
 
+  processPhosphorMessage(msg: any) {
+    this._processLuminoMessage(msg, super["processPhosphorMessage"]);
+
+    var that = this;
+    if (msg.type === "before-attach") {
+      window.addEventListener("resize", function () {
+        that.autosizeFigure();
+      });
+    }
```

Comment:
```suggestion
    var that = this;
    if (msg.type === "before-attach") {
      window.addEventListener("resize", function () {
        that.autosizeFigure();
      });
    }
```
I guess we can remove this?

Reply:
If we want compatibility, probably not? This is not done in `processLuminoMessage`, just in `processPhosphorMessage`

Reply:
I see the same code in `_processLuminoMessage` right? (Modulo the code changes due to https://github.com/plotly/plotly.py/pull/3805)

Reply:
Sorry to ask, but is this related also to 
- #3689

Reply:
I was referring to https://github.com/jupyter-widgets/ipywidgets/pull/3124, but I think with this current PR shrinking actually works fine (worth double checking).

Reply:
Yes, let's please remove those lines, as they are indeed already executed in the `this._processLuminoMessage` call immediately above

Reply:
Otherwise we get two event listeners, right?

Reply:
The shrinking works fine for me too ;) 

But inside an ipywidgets container it stilll does not really resizes correctly. Maybe what @maartenbreddels suggested, applying the  [bqplot/bqplot#1531] strategy would solve the issues.